glide-core/redis-rs/redis/src/cluster_routing.rs

```diff
@@ -287,6 +384,27 @@ fn get_route(is_readonly: bool, key: &[u8]) -> Route {
     }
 }
 
+/// Represents the pattern of argument structures in multi-slot commands,
```

Comment:
Why do we need this? why not use the command metadata - which already has all this information

For example:


```
127.0.0.1:6379> command info mset
1)  1) "mset"
    2) (integer) -3
    3) 1) write
       2) denyoom
    4) (integer) 1
    5) (integer) -1
    6) (integer) 2
...
```

Here:

`4)` `1` is the first key position in the command
`5)` `-1` means that the last key (i.e. can have multiple keys)
`6)` `2` -> step between keys

We should probably cache it during connection initiation and use that info

Reply:
ATM we don't have command information cached and all command information is hardcoded into this file. We have talked about it in the past of moving to dynamic key position / request / response policies, but as you said it is per connection (what happens if we have cluster with nodes in different versions, how do I know which connection metadata should I use to get the key in order to find the key so we'll be able to find the relevant route?) so it can have some implications and to do it we need to have a good design to keep it efficient memory and computation wise (the output of COMMAND is long). So with all of that being said - we decided to keep redis-rs hardcoded routing information and later on we might want to replace it.